pkg/cloudprovider/aws/fleet/nodefactory.go

```diff
@@ -33,7 +33,7 @@ type NodeFactory struct {
 	ec2 ec2iface.EC2API
 }
 
-func (n *NodeFactory) For(ctx context.Context, instanceIds []*string) ([]*v1.Node, error) {
+func (n *NodeFactory) For(ctx context.Context, instanceIds []*string) (map[string]*v1.Node, error) {
```

Comment:
I like this key-ing of node by instanceId. Simplifies code elsewhere, too. Do you mind adding an ultra brief comment on For?

// For a set of instanceIds constructs ...